src/NServiceBus.Core/Transports/Learning/LearningTransportInfrastructure.cs

```diff
@@ -0,0 +1,91 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using DelayedDelivery;
+    using Performance.TimeToBeReceived;
+    using Routing;
+    using Settings;
+    using Transport;
+
+    class LearningTransportInfrastructure : TransportInfrastructure
+    {
+        public LearningTransportInfrastructure(SettingsHolder settings)
+        {
+            this.settings = settings;
+
+            if (!settings.TryGet(StorageLocationKey, out storagePath))
+            {
+                var solutionRoot = FindSolutionRoot();
+                storagePath = Path.Combine(solutionRoot, ".learningtransport");
+            }
+        }
+
+        public override IEnumerable<Type> DeliveryConstraints { get; } = new[]
+        {
+            typeof(DiscardIfNotReceivedBefore),
+            typeof(DelayDeliveryWith),
+            typeof(DoNotDeliverBefore)
+        };
+
+        public override TransportTransactionMode TransactionMode => TransportTransactionMode.SendsAtomicWithReceive;
+
+        public override OutboundRoutingPolicy OutboundRoutingPolicy { get; } = new OutboundRoutingPolicy(OutboundRoutingType.Unicast, OutboundRoutingType.Multicast, OutboundRoutingType.Unicast);
+
+        string FindSolutionRoot()
```

Comment:
Would it be simpler to use the `CallerFilePathAttribute` like https://github.com/Particular/NServiceBus.Persistence.ServiceFabric/blob/develop/src/Tests/R.cs#L229 ?